collections/ansible_collections/purestorage/flashblade/plugins/modules/purefb_fs.py

```diff
@@ -598,6 +598,8 @@ def main():
         delete_fs(module, blade)
     elif state == 'absent' and fsys and fsys.destroyed and module.params['eradicate']:
         eradicate_fs(module, blade)
+    elif state == 'absent' and fsys and fsys.destroyed:
+        module.exit_json(changed=False)
     elif state == 'absent' and not fsys:
         module.exit_json(changed=False)
 
```

Comment:
Add `module.exit_json(changed=False)` here. This will ensure a clean exit on any missed condition in the future.

Reply:
@sdodsley Do you want me to keep only line number 602( module.exit_json(changed=False) and remove line number 601( elif state == 'absent' and fsys and fsys.destroyed:) ?

Reply:
Yeah - remove your newly added lines at 601 and 602

Reply:
@mahendra-maplelabs

Reply:
@sdodsley Ok, But If I remove this condition, Running ansible-playbook with "state: absent" failing with error "New-style module did not handle its own exit" if the filesystem is already deleted but not eradicated.

Reply:
That is why I said add `module.exit_json(changed=False)` after all the `elif` checks around line 604

Reply:
@mahendra-maplelabs

Reply:
@sdodsley Ok, Thanks a lot for the clarification. I will do the change.